conn/conn.go

```diff
@@ -0,0 +1,131 @@
+package conn
+
+import (
+	"net"
+	"time"
+)
+
+type (
+	// RLConnection is a wrapper for the net.Conn object that allows
+	// the user to limit the connection throughput.
+	RLConnection struct {
+		conn             net.Conn  // underlying net.Conn
+		packetSize       uint64    // size of single packet in bytes
+		packetsPerSecond uint64    // packets that are sent per second
+		writtenBytes     uint64    // number of bytes that have been written to the latest packet
+		lastWrite        time.Time // time when last packet was started
+	}
+)
+
+// Dial creates a RateLimitedDialer.
+func Dial(dialer *net.Dialer, network string, address string) (*RLConnection, error) {
+	conn, err := dialer.Dial(network, address)
+	if err != nil {
+		return nil, err
+	}
+	rlc := &RLConnection{
+		conn: conn,
+	}
+	return rlc, nil
+}
+
+// Read calls the underlying connection's Read method.
+func (rlc *RLConnection) Read(b []byte) (n int, err error) {
+	return rlc.conn.Read(b)
+}
```

Comment:
no rate limiting on reads?

Reply:
You can't just review the hell out of my WIP and expect everything to be implemented :P